src/libsystemd-network/sd-dhcp6-client.c

```diff
@@ -197,8 +197,11 @@ int sd_dhcp6_client_set_mac(
                 assert_return(addr_len == ETH_ALEN, -EINVAL);
         else if (arp_type == ARPHRD_INFINIBAND)
                 assert_return(addr_len == INFINIBAND_ALEN, -EINVAL);
-        else
-                return -EINVAL;
+        else {
+                client->arp_type = ARPHRD_NONE;
+                client->mac_addr_len = 0;
+                return 0;
```

Comment:
It still breaks in configuration that works before but at least people can now configure `DUIDType=` to something to `uuid` or `vendor` to make the link not fail.

Reply:
Fixed now.